native/src/main/java/org/ballerinax/asb/listener/MessageListener.java

```diff
@@ -34,46 +34,43 @@
  * ASB message listener representation binding it to a Ballerina service.
  */
 public class MessageListener {
+
     private static final Logger LOGGER = LoggerFactory.getLogger(MessageListener.class);
 
-    private Runtime runtime;
-    private ArrayList<BObject> services = new ArrayList<>();
-    private Map<BObject, MessageDispatcher> dispatcherSet = new HashMap<BObject, MessageDispatcher>();
-    private BObject caller;
-    private ServiceBusClientBuilder sharedConnectionBuilder;
-    private boolean started = false;
+    public static Runtime runtime;
+    private static ArrayList<BObject> services = new ArrayList<>();
+    public static Map<BObject, MessageDispatcher> dispatcherSet = new HashMap<BObject, MessageDispatcher>();
+    private static BObject caller;
+    private static ServiceBusClientBuilder sharedConnectionBuilder;
+    private static boolean started = false;
 
     /**
-     * Initializes Azure Service Bus listener. This creates a connection to the pointed
+     * Initializes Azure Service Bus listener. This creates a connection to the
+     * pointed
      * Azure Service Bus. Actual listeners will get created with the information
      * of attached services.
      *
      * @param connectionString Azure service bus connection string.
+     * @param logLevel         Log level to be used.
+     * @param callerObject     Ballerina object which is used to call the caller.
      */
-    public MessageListener(String connectionString, String logLevel) {
-        this.sharedConnectionBuilder = new ServiceBusClientBuilder().connectionString(connectionString);
+    public static Object initializeListner(String connectionString, String logLevel, BObject callerObject) {
+        sharedConnectionBuilder = new ServiceBusClientBuilder().connectionString(connectionString);
         LOGGER.debug("ServiceBusListnerClient initialized");
+        caller = callerObject;
+        return null;
```

Comment:
Can't we make this as a void method instead of returning null?

Reply:
Sure

Reply:
We return an `Object` because in case of an error we need to return that to the caller. By using `void` it would not work as expected. In here we have not checked whether there is an error or not when initializing the listener. Why have we not checked that here ? AFAIU there can be runtime exceptions when connecting to the ASB.